rewrite-maven/src/main/java/org/openrewrite/maven/OrderPomElements.java

```diff
@@ -0,0 +1,170 @@
+/*
+ * Copyright 2021 the original author or authors.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * https://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.openrewrite.maven;
+
+import org.openrewrite.ExecutionContext;
+import org.openrewrite.Recipe;
+import org.openrewrite.maven.tree.Maven;
+import org.openrewrite.xml.AutoFormat;
+import org.openrewrite.xml.XmlVisitor;
+import org.openrewrite.xml.tree.Content;
+import org.openrewrite.xml.tree.Xml;
+
+import java.util.*;
+
+public class OrderPomElements extends Recipe {
+    private static final List<String> REQUIRED_ORDER = Arrays.asList(
+            "modelVersion",
+            "parent",
+            "groupId",
+            "artifactId",
+            "version",
+            "packaging",
+            "name",
+            "description",
+            "url",
+            "inceptionYear",
+            "organization",
+            "licenses",
+            "developers",
+            "contributors",
+            "mailingLists",
+            "prerequisites",
+            "modules",
+            "scm",
+            "issueManagement",
+            "ciManagement",
+            "distributionManagement",
+            "properties",
+            "dependencyManagement",
+            "dependencies",
+            "repositories",
+            "pluginRepositories",
+            "build",
+            "reporting",
+            "profiles");
+
+    @Override
+    public String getDisplayName() {
+        return "Order POM elements";
+    }
+
+    @Override
+    public String getDescription() {
+        return "Order POM elements according to the [recommended](http://maven.apache.org/developers/conventions/code.html#pom-code-convention) order.";
+    }
+
+    @Override
+    public Set<String> getTags() {
+        return Collections.singleton("RSPEC-3423");
+    }
+
+    @Override
+    protected MavenVisitor getVisitor() {
+        return new MavenVisitor() {
```

Comment:
No need to have a nested visitor here. MavenVisitor IS an XmlVisitor. 

Just override those methods in your MavenVisitor and get rid of OrderVisitor.

Reply:
nice suggestion thanks @tkvangorder